yt_dlp/extractor/pialive.py

```diff
@@ -0,0 +1,126 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    clean_html,
+    extract_attributes,
+    get_element_by_class,
+    get_element_html_by_class,
+    multipart_encode,
+    unified_timestamp,
+    url_or_none,
+)
+from ..utils.traversal import traverse_obj
+
+
+class PiaLiveIE(InfoExtractor):
+    PLAYER_ROOT_URL = 'https://player.pia-live.jp/'
+    PIA_LIVE_API_URL = 'https://api.pia-live.jp'
+    API_KEY = 'kfds)FKFps-dms9e'
+    _VALID_URL = r'https?://player\.pia-live\.jp/stream/(?P<id>[\w-]+)'
+
+    _TESTS = [
+        {
+            'url': 'https://player.pia-live.jp/stream/4JagFBEIM14s_hK9aXHKf3k3F3bY5eoHFQxu68TC6krUDqGOwN4d61dCWQYOd6CTxl4hjya9dsfEZGsM4uGOUdax60lEI4twsXGXf7crmz8Gk__GhupTrWxA7RFRVt76',
+            'info_dict': {
+                'id': '88f3109a-f503-4d0f-a9f7-9f39ac745d84',
+                'display_id': '2431867_001',
+                'title': 'こながめでたい日２０２４の視聴ページ | PIA LIVE STREAM(ぴあライブストリーム)',
+                'live_status': 'was_live',
+                'comment_count': int,
+            },
+            'params': {
+                'getcomments': True,
+                'skip_download': True,
+                'ignore_no_formats_error': True,
+            },
+            'skip': 'The video is no longer available',
+        },
+        {
+            'url': 'https://player.pia-live.jp/stream/4JagFBEIM14s_hK9aXHKf3k3F3bY5eoHFQxu68TC6krJdu0GVBVbVy01IwpJ6J3qBEm3d9TCTt1d0eWpsZGj7DrOjVOmS7GAWGwyscMgiThopJvzgWC4H5b-7XQjAfRZ',
+            'info_dict': {
+                'id': '9ce8b8ba-f6d1-4d1f-83a0-18c3148ded93',
+                'display_id': '2431867_002',
+                'title': 'こながめでたい日２０２４の視聴ページ | PIA LIVE STREAM(ぴあライブストリーム)',
+                'live_status': 'was_live',
+                'comment_count': int,
+            },
+            'params': {
+                'getcomments': True,
+                'skip_download': True,
+                'ignore_no_formats_error': True,
+            },
+            'skip': 'The video is no longer available',
+        },
+    ]
+
+    def _extract_vars(self, variable, html):
```

Comment:
`_extract_vars` is a misnomer if it only extracts one at a time
```suggestion
    def _extract_var(self, variable, html):
```